src/WebJobs.Script/Workers/ProcessManagement/ProcessMonitor.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.ProcessManagement
+{
+    public class ProcessMonitor : IDisposable
```

Comment:
Please add interface for this
- To allow unit testing
- To give us ability to add various implemenations based on hosting environment

Reply:
Added an IProcessMetricsProvider interface that this type uses, which allows for alternate implementations and testing. I've added tests as well.